File: backend/src/main/java/com/kamiokaweb/redo/model/taskitem/TaskItem.java
package com.kamiokaweb.redo.model.taskitem;

import com.kamiokaweb.redo.model.item.Item;
import io.swagger.v3.oas.annotations.media.Schema;

@Schema(description = "依頼明細")
public record TaskItem(
        TaskItemId taskItemId,
        @Schema(description = "品目") Item item,
        Quantity quantity,
        Amount amount
) {
}

File: frontend/types/taskitem.ts
import { Item } from "./item";

export interface TaskItemId {
  value: number;
}

export interface Quantity {
  value: number;
}

export interface Amount {
  value: number;
}

export interface TaskItem {
  taskItemId: number | null;  // @JsonValueのため直接数値
  item: Item;
  quantity: number;  // @JsonValueのため直接数値
  amount: number;  // @JsonValueのため直接数値
}
